tests/snuba/sessions/test_sessions_v2.py

```diff
@@ -493,6 +515,125 @@ def test_massage_virtual_groupby_timeseries():
     assert actual_result == expected_result
 
 
+@freeze_time("2020-12-18T13:25:15.769Z")
+def test_clamping_in_massage_sessions_results_with_groupby_timeseries():
+    query = _make_query(
+        "statsPeriod=18h&interval=6h&field=sum(session)&field=count_unique(user)&groupBy=session.status"
+    )
+    # snuba returns the datetimes as strings for now
+    result_timeseries = [
+        {
+            "sessions": 5,
+            "sessions_errored": 10,
+            "sessions_crashed": 0,
+            "sessions_abnormal": 0,
+            "users": 5,
+            "users_errored": 10,
+            "users_crashed": 0,
+            "users_abnormal": 0,
+            "bucketed_started": "2020-12-18T00:00:00+00:00",
+        },
+        {
+            "sessions": -4,
+            "sessions_errored": -4,
+            "sessions_crashed": -4,
+            "sessions_abnormal": 0,
+            "users": -4,
+            "users_errored": -4,
+            "users_crashed": -4,
```

Comment:
I don’t think snuba will ever return negative numbers here. Not sure if we should test that case, having unbalanced total/errored counts might be enough.

Reply:
The point of this test is to test that all session counts are clamped to 0 if they are provided as negative according to a conversation I had with @jan-auer .. you don't think that is necessary to test that behaviour?

Reply:
I believe the majority of our integer columns in clickhouse are unsigned ints so snuba won't be able to return negative numbers

Reply:
Thanks for the clarification @wmak @Swatinem